src/Nethermind/Nethermind.Blockchain/ChainHeadReadOnlyStateProvider.cs

```diff
@@ -0,0 +1,74 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Int256;
+using Nethermind.State;
+using Nethermind.Trie;
+
+namespace Nethermind.Blockchain
+{
+    public class ChainHeadReadOnlyStateProvider : IReadOnlyStateProvider
```

Comment:
all good, I am just thinking about two things:

how to avoid having two different interfaces -> IStateReder and IReadOnlySTateProvider
it seems to be increasing complexity a bit if both classes stay

---

second thing - it would be great to have a regression test here so we can see what problem we solve

Reply:
Hard to add test as its setup/configuration only.
For now I wanted minimal compatible changes - hece extracting smaller interface, don't really have a better idea but we can change it later.